stdlib/universe/join.go

```diff
@@ -918,30 +930,55 @@ func (c *MergeJoinCache) join(left, right *execute.ColListTableBuilder) (flux.Ta
 					leftRecord := left.GetRow(l)
 					rightRecord := right.GetRow(r)
 
+					var err error
 					leftRecord.Range(func(columnName string, columnVal values.Value) {
 						column := tableCol{
 							table: c.names[c.leftID],
 							col:   columnName,
 						}
-						newColumn := c.schemaMap[column]
-						newColumnIdx := c.colIndex[newColumn]
+						newColumn, ok := c.schemaMap[column]
+						if !ok {
+							println("left Column not found in schema")
+							err = errors.Newf(codes.Internal, "column '%s' not found in join schema", columnName)
+							return
+						}
+						newColumnIdx, ok := c.colIndex[newColumn]
+						if !ok {
+							println("left index not found in map")
+							err = errors.Newf(codes.Internal, "could not find index for column '%s' in column index map", columnName)
+						}
```

Comment:
I wonder if an internal error is what we really want here. Users don't get to see internal error messages, so they won't know how to fix their query when they hit this. It's a little weird to make it a user error since the user is not really at fault here. I wish there was a `not yet implemented` message.

Reply:
I agree that it's a weird situation. I originally had this as `codes.Invalid` thinking that we would want to show an error to the user. After discussing it with @nathanielc though, we came to the conclusion that, because the user isn't at fault here, logging an internal error and letting the table be discarded was the best approach until we can resolve #4315. I'm open to having my mind changed on that though.

Reply:
Thanks for the rational, I think that makes sense.